packages/libs/chainwebjs/src/tests/transactions.test.ts

```diff
@@ -0,0 +1,184 @@
+/* eslint-disable @typescript-eslint/naming-convention */
+
+jest.mock('cross-fetch', () => {
+  return {
+    __esModule: true,
+    default: jest.fn(),
+  };
+});
+
+import { mockFetch } from './mokker';
+import { filterTxs } from '../transactions';
+import {
+  filterData,
+  filterDataNoTx,
+  filterDataFormatted,
+} from './mocks/recentsfilterDataMock';
+import { IBlockPayloads, ITransactionElement } from '../types';
+
+import fetch from 'cross-fetch';
+
+const mockedFunctionFetch = fetch as jest.MockedFunction<typeof fetch>;
+mockedFunctionFetch.mockImplementation(
+  mockFetch as jest.MockedFunction<typeof fetch>,
+);
+import chainweb from '..';
+/* ************************************************************************** */
+/* Test settings */
+
+jest.setTimeout(25000);
+const debug: boolean = false;
+
+/* ************************************************************************** */
+/* Test Utils */
+
+const logg = (...args: unknown[]): void => {
+  if (debug) {
+    console.log(...args);
+  }
+};
+
+/* ************************************************************************** */
+/* Blocks */
+
+const header = {
+  adjacents: {
+    '10': 'oT8NLW-IZSziaOgI_1AfCdJ18u3epFpGONrkQ_F6w_Y',
+    '15': 'ZoBvuaVZWBOKLaDZfM51A9LaKb5B1f2fW83VLftQa3Y',
+    '5': 'SDj4sXByWVqi9epbPAiz1zqhmBGJrzSY2bPk9-IMaA0',
+  },
+  chainId: 0,
+  chainwebVersion: 'mainnet01',
+  creationTime: 1617745627822054,
+  epochStart: 1617743254198411,
+  featureFlags: 0,
+  hash: 'BsxyrIDE0to4Kn9bjdgR_Q7Ha9bYkzd7Yso8r0zrdOc',
+  height: 1511601,
+  nonce: '299775665679630368',
+  parent: 'XtgUmsnF20vMX4Dx9kN2W8cIXXiLNtDdFZLugMoDjrY',
+  payloadHash: '2Skc1JkkBdLPkj5ZoV27nzhR3WjGD-tJiztCFGTaKIQ',
+  target: '9sLMdbnd1x6vtRGpIw5tKMt_1hgprJS0oQkAAAAAAAA',
+  weight: 'iFU5b59ACHSOAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA',
+};
+
+const height = header.height;
+const blockHash = header.hash;
+
+/* ************************************************************************** */
+/* By Height */
+
+describe('by height', () => {
+  test('Transactions', async () => {
+    const r = await chainweb.transaction.height(0, height);
+    logg('Transactions:', r);
+    expect(r.length).toBe(1);
+    expect(r[0]).toHaveProperty('transaction');
+    expect(r[0]).toHaveProperty('output');
+    expect(r[0]).toHaveProperty('height');
+    expect(r[0].height).toBe(height);
+  });
+});
+
```

Comment:
The describes in this file should probably be wrapped in a

`describe('chainweb.transactions`, () => {`

with `it`s being what is now in the current `describe`s